### PR TITLE
Enhance CPU simulator logging and throughput metrics

### PR DESCRIPTION
**Pull request by yuxin00j**
### Description
This PR adds logging enhancements and throughput calculation fixes to the macrobenchmark CPU simulator in `gcsfs`.

### Key Changes
- **Checkpoint Overhead Exclusion**: Checkpointing duration (both saving and removing) is now tracked and deducted from `step_time` to prevent skewed throughput numbers during checkpoint steps.
- **Accurate Data Loading Metrics**: The step timer is initialized at `on_train_epoch_start` to ensure the first batch captures the initial data loading delay.
- **Detailed Throughput Logging**: Emits both `local_throughput` and `global_throughput` per optimizer step.
- **Targeted Profiling Hooks**: Adds profiler hooks to isolate and measure `FitLoop.setup_data` and `_PrefetchDataFetcher.__iter__` (worker spawn times).
- **Dataset Load Timing**: Injects a timer around `datasets.load_dataset` to measure HF dataset preparation time.

### CODE REVIEW

File: gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py [hunk 1]
```diff
@@ -294,24 +296,37 @@ def on_train_batch_end(self, trainer, pl_module, outputs, batch, batch_idx):
...
-        pl_module.log("throughput", throughput)
+        pl_module.log("local_throughput", local_throughput)
+        pl_module.log("global_throughput", global_throughput)
 
         loss = outputs["loss"] if isinstance(outputs, dict) else outputs
         loss_val = loss.item() if isinstance(loss, torch.Tensor) else loss
         logging.info(
             "Global Rank: %d | Step: %d | Loss: %.4f | Step Time: %.4fs | "
-            "Throughput: %.2f samples/s",
+            "Throughput: %.2f samples/s | Local Throughput: %.2f samples/s",
```
Comment (zhixiangli) [inline gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py:317]: Will this break the regex used to parse the metric? Could you confirm?

File: gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py [hunk 2]
```diff
@@ -517,5 +548,47 @@ def build_strategy(name):
     else:
         checkpoint_load_path = None
 
+    # TODO: These are underscore private APIs, which may break during a Lightning upgrade.
```
Comment (zhixiangli) [inline gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py:551]: Could you please create a PR for PyTorch Lightning and add the link in the comments? We can clean this up once that PR is merged.

File: gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py (hunk 1 above)
Comment (yuxin00j) [reply gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py:317]: It won't break since the part before the 'Local throughput' will match the regex https://github.com/fsspec/gcsfs/blob/main/cloudbuild/macrobenchmarks/metrics/parsers/hf.py#L18.

File: gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py (hunk 2 above)
Comment (yuxin00j) [reply gcsfs/tests/perf/macrobenchmarks/workloads/hf-pytorch-lightning-cpu/helm_chart/llama_3_1_8b_cpu_sim.py:551]: Done.
